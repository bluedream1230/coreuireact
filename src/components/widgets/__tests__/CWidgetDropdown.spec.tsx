import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CWidgetDropdown } from '../../../index'

test('loads and displays CWidgetDropdown component', async () => {
  const { container } = render(<CWidgetDropdown/>)
  expect(container).toMatchSnapshot()
})

test('CWidgetDropdown customize', async () => {
    const { container } = render(
        <CWidgetDropdown
            className="bazinga"
            action="action"
            change="change"
            chart="chart"
            color="info"
            title="title"
            value="value"
        >
            Test
        </CWidgetDropdown>
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('bg-info')
    expect(container.firstChild).toHaveClass('text-high-emphasis-inverse')
    expect(container.firstChild).toHaveClass('bazinga')
    if(container.firstChild === null){
        expect(true).toBe(false)
    }else{
        expect(container.firstChild.firstChild).toHaveClass('pb-0')
        expect(container.firstChild.firstChild).toHaveClass('d-flex')
        expect(container.firstChild.firstChild).toHaveClass('justify-content-between')
        expect(container.firstChild.firstChild).toHaveClass('align-items-start')
        if(container.firstChild.firstChild === null || container.firstChild.firstChild.firstChild === null){
            expect(true).toBe(false)
        }else{
            expect(container.firstChild.firstChild.firstChild.firstChild).toHaveClass('fs-4')
            expect(container.firstChild.firstChild.firstChild.firstChild).toHaveClass('fw-semibold')
            if(container.firstChild.firstChild.firstChild.firstChild === null){
                expect(true).toBe(false)
            }else{
                expect(container.firstChild.firstChild.firstChild.firstChild.lastChild).toHaveClass('fs-6')
                expect(container.firstChild.firstChild.firstChild.firstChild.lastChild).toHaveClass('fw-normal')
                expect(container.firstChild.firstChild.firstChild.firstChild.lastChild).toHaveTextContent('change')
            }
        }
    }

    //expect(container.firstChild).toHaveTextContent('Test')
})