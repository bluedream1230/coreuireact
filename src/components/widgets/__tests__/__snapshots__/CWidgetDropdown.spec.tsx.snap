// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidgetDropdown customize 1`] = `
<div>
  <div
    class="card bg-info text-high-emphasis-inverse bazinga"
  >
    <div
      class="card-body pb-0 d-flex justify-content-between align-items-start"
    >
      <div>
        <div
          class="fs-4 fw-semibold"
        >
          value
           
          <span
            class="fs-6 fw-normal"
          >
            change
          </span>
        </div>
        <div>
          title
        </div>
      </div>
      action
    </div>
    chart
  </div>
</div>
`;

exports[`loads and displays CWidgetDropdown component 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-body pb-0 d-flex justify-content-between align-items-start"
    >
      <div />
    </div>
  </div>
</div>
`;
